### ISSUE TITLE
[question] Boost_<COMPONENT>_Found cmake variables missing ?

### ISSUE THREAD

**Issue by ggulgulia** [conan-io/conan-center-index#11085]
Hello there,

##  Description
* I'm creating conan recipe for [OGRE](https://github.com/OGRECave/ogre-next) which is an important graphics rendering engine for many applications like games and simulations, including ignition-gazebo etc. 
* OGRE relies on boost and the standard boost cmake variables most like `[Boost_THREAD_FOUND](https://github.com/OGRECave/ogre-next/blob/49653fc931d923b5f0c323d48b965e38f0d64525/CMake/Packages/FindOGRE.cmake#L329)` , `Boost_DATE_TIME_FOUND` etc .
* The conan generated `FindBoost.cmake` doesn't seem to provide the cmake variables because of which the build doesn't work as expected

## Comparison with official `FindBoost.cmake`  cmake documentation
* The official cmake documentation for `[FindBoost.cmake](https://cmake.org/cmake/help/latest/module/FindBoost.html#result-variables)` sets these variables . 
* I also tried building OGRE with system installed boost and it worked fine  since the **system installed** `FindBoost.cmake` sets these variables. See line 312 in the attached file (extension changed)
[FindBoost.cmake.txt](https://github.com/conan-io/conan-center-index/files/8858849/FindBoost.cmake.txt)
 
## Question
* Does conan recipe for boost provide these variables or any variable to verify if a component was found ?
* If not, will CCI consider adding these cmake variables ?
* Is there a recommended short term fix or a recipe that I can look into from another recipe and with which I can work now ?

**Comment by prince-chrismc** [comment]
> doesn't seem to provide the cmake variables because of which the build doesn't work as expected

You are correct they are not provided, I was trying to see if there a reason the might have been mentioned but I as not able.

> Does conan recipe for boost provide these variables or any variable to verify if a component was found ?

I have two paths as a community reiviewer

- Using if checks in the CMake for targets `if(TARGET Boost::thread)`
- Validate the components in the conanfile and assume they are present in CMake

the second option is prefered, since Conan can ensure the correct package is used,

https://github.com/conan-io/conan-center-index/blob/10c8d4778344af57664c0acb89285acfac89e98f/recipes/wt/all/conanfile.py#L98

https://github.com/conan-io/conan-center-index/blob/9a6866e3cc3e6e9795ed675c8fc3da328133cb1c/recipes/gtsam/all/patches/0001-conan-4.0.2.patch#L60

**Comment by ggulgulia** [comment]
The second option seems correct but in the case where the components are searched on CMake level like in [OGRE](https://github.com/OGRECave/ogre-next/blob/49653fc931d923b5f0c323d48b965e38f0d64525/CMake/Packages/FindOGRE.cmake#L329),

```
 if (NOT Boost_THREAD_FOUND)
        set(OGRE_DEPS_FOUND FALSE)
 else ()
 ...
```
  will the use of second option set the variable for CMake for build to work correctly  or do we still need patches to fix the `Boost_<COMPONENT>_FOUND` variables ?
